components/Panel/StakeUnstakePanel/StakeUnstakePanel.tsx

```diff
@@ -64,7 +63,13 @@ export function StakeUnstakePanel() {
   }
 
   function stake(stakeAmountInput: string, resetStakeAmount: () => void) {
-    const stakeAmount = parseEtherSafe(stakeAmountInput);
+    if (!unstakedBalance) return;
+    const parsedStakeAmountInput = parseEtherSafe(stakeAmountInput);
```

Comment:
handle potentially too large number input when using max

Reply:
I'm trying to find out if there might not be a deeper problem that makes `parsedStakeAmountInput.gt(unstakedBalance)===true`

Do you think this treatment could be affecting the accuracy of the numbers?
[parseEtherSafe](https://github.com/UMAprotocol/voter-dapp-v2/blob/37caf4e710238a123ea54450fb087d85c4cd300a/helpers/web3/ethers.ts#L11)

Have you tried using here, instead of `parseEtherSafe`, ethers like:
`const parsedStakeAmountInput = ethers.utils.parseEther(stakeAmountInput);`

Reply:
Yeah that could be, since there's some casting happening there. I'll try this and let you know

Reply:
You were right 👍